server.js

```diff
@@ -21,10 +16,30 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+	res.json({BooksClubMessage: " Welcome to API for book data!", data: listEndpoints(app)});
 });
 
-// Start the server
+//Create a new endpoint with collection of all Books data
+app.get("/books", (req, res) => {
+	res.status(200).json({booksData: booksData, success: true});
+});
+//Create a new endpoint with authors in alphabetic order
+app.get("/books/authors/", (req, response) => {
+  const allAuthors = booksData.map((item) => item.authors).sort();
+	console.log(allAuthors);
+	response.status(200).json(allAuthors);
+});
+//Create a new endpoint with bookID
+app.get("/books/:id", (req, res) => {
+  const id = req.params.id
+  const bookId = booksData.find((item) => item.bookID === +id)
+//404
+  if (!bookId) {
+    res.status(404).json({ errorMessage: "No book with this id found. Try to find the right id" })
+  }
+  res.json(bookId)
+})
+// Start the server.
```

Comment:
This one worked as well.
A nice touch i would like to add is if a user types in wrong ID, you can add the ID in the error message (whatever they had put in)....

Const { id } = req.params;

On the errorMessage: "No book with this id ${id} found. Try to find....."

Reply:
det ska vara ` ` istället för ""